google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/project/AppEngineProjectService.java

```diff
@@ -101,4 +104,29 @@ public abstract AppEngineStandardRuntime getAppEngineStandardDeclaredRuntime(
    */
   public abstract boolean isJarOrWarMavenBuild(@NotNull Module module);
 
+  public abstract Optional<String> getServiceNameFromAppYaml(@NotNull String appYamlPath);
+
+  public abstract FlexibleRuntime getFlexibleRuntimeFromAppYaml(@NotNull String appYamlPathString);
+
+  /**
+   * Gets the service specified in an appengine-web.xml file, in its first found service or module
+   * tag. Service has precedence over module.
+   *
+   * @return the value of the first found service tag, or else the value of the first found module
+   * tag, or else "default"
+   */
+  public abstract String getServiceNameFromAppEngineWebXml(
```

Comment:
deploymentSource is to extract the IntelliJ module. I added some Javadoc, hopefully it clears things up.